webgui-new/.eslintrc.json

```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "next/core-web-vitals"
```

Comment:
Is this enough? Does this include React specific linting rules? Maybe `plugin:react/recommended` should be added as well.

Reply:
They have been added. See the `eslint.config.js` file. The most important React rules are also configured by default by the plugin. The `.eslintrc.json` file was generated automatically when running `lint`. I'm assuming it is required for Next.js.

Reply:
I see, I was not familiar with this *new* ESLint configuration file format.
https://eslint.org/docs/latest/use/configure/configuration-files-new

Then, it is fine, but I think, the old config file (`.eslintrc.json`) should be removed.
https://eslint.org/docs/latest/use/configure/configuration-files

It is strange that there are 2 config files (old and new format) with different content. The `next/core-web-vitals `ruleset` should be added to the new config file as well I guess.

Reply:
> It seems Next.js can only use the .eslintrc.json config file, because it was automatically created when I ran lint. The new eslint.config.js file was overriden by this file, and I couldn't see the errors. So I added the react plugin to the .eslintrc.json file, and removed the new config file.

Good thing this turned out in time! The existence of 2 config files was strange fortunately 😄 

> This way, the errors are visible. There were errors in almost every file, because ESLint did not recognize React and the JSX namespace. As a fix I had to import 'React', and remove the 'JSX.Element' type enforcement (it was not required anyway, because they are already set to that type).

Cannot this be fixed by adding `"JSX": true` to the `globals`? See e.g. here:
https://gitlab.com/tms-elte/frontend-react/-/blob/412cf5a123249db0b0ddc4585743050f5385bdf6/.eslintrc.json#L60
I think that should be the real fix, not removing the valid type enforcement from the code!

Also maybe `@typescript-eslint` and `plugin:`react-hooks` should be considered for linting. (Or some of their rules.)

Reply:
Yes I tried setting `globals` before, apparently, I set `jsx` and not `JSX`, now it works, thanks for providing an example. I added back the `JSX.Element` type enforcements, and also added the `@typescript-eslint` and `react-hooks` plugins and fixed the warnings related to them.